src/app/store/slices/storage/storage.thunks/uploadFolderThunk.ts

```diff
@@ -296,6 +307,21 @@ function countItemsUnderRoot(root: IRoot): number {
 
   return count;
 }
+function getItemsSize(root: IRoot): number {
+  let size = 0;
+
+  const queueOfFolders: Array<IRoot> = [root];
+
+  while (queueOfFolders.length) {
+    const folder = queueOfFolders.shift() as IRoot;
+
+    size += folder.childrenFiles?.map((file) => file.size).reduce((a, b) => a + b, 0) ?? 0;
```

Comment:
why not just: 

`folder.childrenFiles?.forEach((file) => size += file.size);`

?